File: sem3/map/ToyLanguageInterpreter/src/utils/FileTable.java
package utils;

import java.util.Map;

/**
 * Created by mirko on 12/10/2016.
 */
public interface FileTable<K, V> {
    void add(K key, V value);

    V remove(K key);

    boolean contains(K key);

    V getValue(K key);

    String toString();

    Iterable<Map.Entry<K, V>> getAll();
}

File: sem3/map/ToySharpLanguageInterpreter/ToySharpLanguageInterpreter/src/utils/FileTable.cs
using System;
using System.Collections.Generic;

namespace ToySharpLanguageInterpreter
{
	public interface FileTable<K, V>
	{
		void add(K key, V value);

		V remove(K key);

		bool contains(K key);

		V getValue(K key);

		string ToString();

		List<KeyValuePair<K, V>> getAll();
	}
}